language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToFunctionCodeAction.java

```diff
@@ -403,6 +461,42 @@ private String getFunctionName(CodeActionContext context, NonTerminalNode matche
         return NameUtil.generateTypeName(EXTRACTED_PREFIX, visibleSymbolNames);
     }
 
+    private List<NonTerminalNode> getPossibleExpressionNodes(NonTerminalNode node, 
+                                                             ExpressionNodeValidator nodeValidator) {
+        // Identify the sub-expressions to be extracted
+        List<NonTerminalNode> nodeList = new ArrayList<>();
+        while (node != null && !isExtractableExpressionNode(node) && node.kind() != SyntaxKind.OBJECT_FIELD
+                && !nodeValidator.getInvalidNode()) {
+            nodeList.add(node);
+            node = node.parent();
+            node.accept(nodeValidator);
+        }
+        return nodeList;
+    }
+
+    private static boolean isExtractableExpressionNode(NonTerminalNode node) {
```

Comment:
The name should be something like `isStatamentNode`( Based on the usage on line 468)